copy_1.py

```diff
@@ -0,0 +1,70 @@
+from random import randint
+
+Hidden_Pattern=[[' ']*8 for x in range(8)]
+Guess_Pattern=[[' ']*8 for x in range(8)]
+
+let_to_num={'A':0,'B':1, 'C':2,'D':3,'E':4,'F':5,'G':6,'H':7}
+
+def print_board(board):
+    print(' A B C D E F G H')
+    print(' ***************')
+    row_num=1
+    for row in board:
+        print("%d|%s|" % (row_num, "|".join(row)))
+        row_num +=1
+
+def get_ship_location():
+    #Enter the row number between 1 to 8
+    row=input('Please enter a ship row 1-8 ').upper()
+    while row not in '12345678':
+        print("Please enter a valid row ")
+        row=input('Please enter a ship row 1-8 ')
+    #Enter the Ship column from A TO H
+    column=input('Please enter a ship column A-H ').upper()
+    while column not in 'ABCDEFGH':
+        print("Please enter a valid column ")
+        column=input('Please enter a ship column A-H ')
+    return int(row)-1,let_to_num[column]
```

Comment:
this function works but can refactor to cancel out the code repetition. we can actually split the functionality here and divvy them out into 2 separate functions. something like this for measure

`
def get_valid_input(prompt, valid_values):

while True:
    value = input(prompt).upper()
    if value in valid_values:
        return value
    print("Please enter a valid value.")


def get_ship_location():

row = get_valid_input('Please enter a ship row 1-8 ', '12345678')
column = get_valid_input('Please enter a ship column A-H ', 'ABCDEFGH')
return int(row) - 1, letter_to_number[column]


`